doc/timeline.md

```diff
@@ -0,0 +1,205 @@
+# Timeline And Keyframes
+
+The `Ra::GuiBase::Timeline` class provides display and management of `Ra::Core::Animation::KeyFramedValue`s.
+
+`Ra::Core::Animation::KeyFramedValue`s must be registered into the `Ra::GuiBase::Timeline`, which can be done using the
+`Ra::GuiBase::Timeline::registerKeyFramedValue` methods.
+
+We here consider two types of `Ra::Core::Animation::KeyFramedValue`:
+
+## Static KeyFramedValue:
+Those are `Ra::Core::Animation::KeyFramedValue`s that are an explicit part of a `Ra::Engine::Entity`, `Ra::Engine::Component` or
+`Ra::Engine::RenderObject` data, either filled upon construction or through the `Ra::GuiBase::Timeline`.
+Static `Ra::Core::Animation::KeyFramedValue`s must be registered in the `Ra::GuiBase::Timeline` after the object's
+construction.
+They usually are not bound to an *UpdateCallback* function since the object they
+belong to usually calls `Ra::Core::Animation::KeyFramedValue::at` to query the current value.
+
+Example Usage:
+```c++
+using KeyFrames = Ra::Core::Animation::KeyFramedValue<Scalar>;
+
+struct MyComponentWithKeyFrame : public Ra::Engine::Component
+{
+   MyComponentWithKeyFrame( const std::string& name, Ra::Engine::Entity* entity, Scalar value = 0_ra )
+       : Ra::Engine::Component( name, entity )
+       , m_keyframes( value, 0_ra )
+       , m_currentValue( value )
+   {
+       // use the linear interpolator instead of the default one
+       m_keyframes.setInterpolator( Ra::Core::Animation::interpolate<Scalar> );
+   }
+
+   void initialize() override {}
+   void goTo( Scalar t ) {
+       // Here we use the value at time t.
+       m_currentValue = m_keyframes.at( t );
+   }
+   void update() {
+       // Here we can update other data that depend on the value.
+   }
+
+   KeyFrames m_keyframes;
+   Scalar m_currentValue;
+};
+
+struct MyTimeDependantSystem : public Ra::Engine::AbstractTimedSystem
+{
+   MyTimeDependantSystem( Ra::GuiBase::Timeline* timeline )
+       : Ra::Engine::AbstractTimedSystem()
+       , m_timeline( timeline )
+   {}
+
+   void generateTasks( Ra::Core::TaskQueue* taskQueue, const Ra::Engine::FrameInfo& ) override {
+       // Update all Components data.
+       for ( auto compEntry : m_components )
+       {
+           auto comp = static_cast<MyComponentWithKeyFrame*>( compEntry.second );
+           auto func = std::bind( &MyComponentWithKeyFrame::update, comp );
```

Comment:
Shouldn't be safer to test if `comp` is not null ?

Reply:
In this case, the System only manages `MyComponentWithKeyFrame`s, thus the code is safe.
Would you still test for it? Just to tell users that it might be an issue for their own system?